Compiler.cpp

```diff
@@ -33,15 +40,44 @@ Value *Compiler::emitVar(
     }
     AllocaInst *Alloca = CreateEntryBlockAlloca(TheFunction, VarName);
     Builder->CreateStore(InitVal, Alloca);
+    NamedValues.back()[VarName] = Alloca;
```

Comment:
I think we should check if NamedValues does have any items here, might throw you never know with C++ really.